### ISSUE TITLE
File extensions prevent grouping

### ISSUE THREAD

**Issue by nwidynski** [kairome/eslint-plugin-grouped-import#3]
I'm trying to group internal/relative imports into their own group, like this:

```ts
'@grouped/group': [
  ERROR, { Internal: [ { path: './' } ] },
],
```

Unfortunately, this does not work on relative paths with a file extension (e.g. `./package.json`), because of the regex test in `similarImport`. Is there a specific point why this regex check exists in the first place?

**Comment by kairome** [comment]
yes, it's used to check for imports with extensions and group them separately from other similar paths.

For example, if I have a rule set like this and no regex test

```
    'css': [{ path: '.css' }],
    'components': [{ path: 'components' }],
```

the following code will not be grouped properly

```
// components
import Comp from 'components/Comp';

// css
import s from 'components/Comp.css';
```

Unfortunately, I can't offer you any solution to your problem for now, other than grouping imports based on file extensions

If you have any ideas, let me know =)

**Comment by nwidynski** [comment]
@kairome Maybe using regex matching instead of includes? That way i might be able to craft a pattern that doesnt match extensions.

**Comment by kairome** [comment]
@nwidynski can you please provide me with your config and an example of relative imports you have?

**Comment by nwidynski** [comment]
@kairome 

```tsx
  'import-group/group': [
    'error',
    { Internal: [{ path: './' }] },
  ],
```

```tsx
import maxSpecifiers from './rules/max-specifiers';
```

**Comment by kairome** [comment]
@nwidynski I published the new version, please let me know if it works now

**Comment by lukascivil** [comment]
I think that only with regex will it work well, it is a way for the tool to deliver all its potential to the developer.

My case is similar @kairome:
```
'import-group/group': [
    'error',
    { Packages: [{ path: 'danger' }] },
    { Modules: [{ path: '@/' }] },
  ],
```

and then: 

![Image](https://github.com/user-attachments/assets/93439754-e404-4542-a067-22a965277f70)

It was expected to pass.
![Image](https://github.com/user-attachments/assets/1549a7a9-baac-456a-bbab-a5d33b1b8a93)

Got into a substring case, how to solve this rule concurrency problem?
